### DISCUSSION TITLE
Add a specific treatment for a color

### DISCUSSION THREAD

**Discussion by bepolymathe** [aidenlx/obsidian-zotero#217] [Help]
Hi, 

Thank you for this wonderful plugin and the community behind it. I'm very pleased with the use of a template for classifying annotations by type, with each type corresponding to a color. 

For `zt-annots-eta `

```
<% const byColor = Object.groupBy(it, (annot) => annot.colorName);
const label = {
  "yellow": "Informations 🟡",
  "red": "Désaccords 🔴",
  "green": "Apports 🟢",
  "blue": "Questions 🔵",
  "purple": "Définitions 🟣",
  "magenta": "Personnes 🧐",
  "orange": "Références 🟠",
  "gray": "Méthodologie 📊",
};
// Merge colors with customized label with unexpected colors, if any
// Keep the order of the colors from the original color-label map
const colorSet = new Set([...Object.keys(label), ...Object.keys(byColor)]);
for (const color of colorSet) { 
if (!(color in byColor)) continue -%>

### <%= label[color] ?? color %>
  <%_ for (const annot of byColor[color]) { %>

<%~ include("annotation", annot) %>
  <%_ } %>
<% } %>
```

However, I'd like to add something else but I'm having trouble figuring out how. I'd like the "magenta" color (and only that one) to insert the annotation between double brackets. I only use this color to highlight the name of an author or character mentioned in an article/book. So it would be very useful if this were inserted in Obsidian between double brackets to have it in the links. 

Could you help me do this?

**Comment by bepolymathe** [comment]
Hi here, 

Could anyone help me with this task?  I'm struggling a bit... (@Quorafind)

**Reply by VolkovIlia** [reply]
for `zt-annot.eta` you should try:

```js
[!note] Page <%= it.pageLabel %>

<%= it.imgEmbed %><mark style="<%- if (it.color) { _%> background-color: <%= it.color %>; <%_ } -%>"><%= it.text %></mark> 
<% if (it.comment) { %>
---
<%= it.comment %>
<% } %>
```

**Comment by aidenlx** [comment]
Direct in v2's Liquid annotation template — `zt.colorName` is on every annotation, so branch on it:

```liquid
{% if zt.colorName == "magenta" %}[[{{ zt.text }}]]{% else %}{{ zt.text }}{% endif %}
```

Palette names are `yellow`, `red`, `green`, `blue`, `purple`, `magenta`, `orange`, `gray`, `plum`. A custom color outside the palette reads as `null`, with the raw value still on `zt.colorHex`.

v2 beta is available now: https://zotlit.aidenlx.site/docs/install-zotlit
Migration guide: https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
